src/pages/index.js

```diff
@@ -100,29 +164,116 @@ function openModal(modal) {
 function closeModal(modal) {
   modal.classList.remove("modal_opened");
   document.removeEventListener("keydown", handleEscKeyPress);
+
+  if (modal === deleteModal) {
+    cardToDelete = null;
+  }
+}
+
+const deleteModalCancelButton = deleteModal.querySelector(
+  ".modal_delete-card_cancel"
+);
+if (deleteModalCancelButton) {
+  deleteModalCancelButton.addEventListener("click", () => {
+    closeModal(deleteModal);
+  });
 }
 
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = nameInput.value;
-  profileDescription.textContent = descriptionInput.value;
-  closeModal(editModal);
+  setButtonText(profileSubmitButton, true, "Saving...");
+
+  api
+    .editUserInfo({
+      name: nameInput.value,
+      about: descriptionInput.value,
+    })
+    .then((user) => {
+      profileName.textContent = user.name;
+      profileDescription.textContent = user.about;
+      editFormElement.reset();
+      resetValidation(
+        editFormElement,
+        [nameInput, descriptionInput],
+        validationConfig
+      );
+      closeModal(editModal);
+      disableButton(
+        editFormElement.querySelector(".modal__submit-button"),
+        validationConfig
+      );
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      setButtonText(profileSubmitButton, false, "Save");
+    });
 }
 
 function handleAddCardSubmit(evt) {
   evt.preventDefault();
-  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
-  const cardElement = getCardElement(inputValues);
-  cardsList.prepend(cardElement);
-  cardForm.reset();
-  disableButton(cardSubmitButton, settings);
-  closeModal(cardModal);
+  setButtonText(cardSubmitButton, true, "Saving...");
+
+  const cardData = {
+    name: cardNameInput.value,
+    link: cardLinkInput.value,
+  };
+  api
+    .addNewCard(cardData)
+    .then((card) => {
+      const cardElement = getCardElement(card);
+      cardsList.prepend(cardElement);
+      cardForm.reset();
+      disableButton(
+        cardForm.querySelector(".modal__submit-button"),
```

Comment:
`cardForm.querySelector(".modal__submit-button"),`


You've already found the submit button of the card form as `cardSubmitButton`


```js
      disableButton(cardSubmitButton, validationConfig);


```